From 87a555dfa338bf2ef27e810b5950de6f929f62c5 Mon Sep 17 00:00:00 2001
From: Tomas Kopecek <tkopecek@redhat.com>
Date: Jul 06 2019 07:43:45 +0000
Subject: [PATCH 1/2] set module_hotfixes=1 in yum.conf via tag config


New field in tag extra 'mock.yum.module_hotfixes' can be used to set
module_hotfixes=0/1 in yum.conf's inside mock's buildroots.

Fixes: https://pagure.io/koji/issue/1521

---

diff --git a/builder/kojid b/builder/kojid
index 0024e4f..8ee15af 100755
--- a/builder/kojid
+++ b/builder/kojid
@@ -272,6 +272,8 @@ class BuildRoot(object):
         opts['target_arch'] = self.target_arch
         if 'mock.package_manager' in self.config['extra']:
             opts['package_manager'] = self.config['extra']['mock.package_manager']
+        if 'mock.yum.module_hotfixes' in self.config['extra']:
+            opts['module_hotfixes'] = self.config['extra']['mock.yum.module_hotfixes']
         if self.internal_dev_setup is not None:
             opts['internal_dev_setup'] = bool(self.internal_dev_setup)
         output = koji.genMockConfig(self.name, self.br_arch, managed=True, **opts)
diff --git a/koji/__init__.py b/koji/__init__.py
index 6a9005b..e9a3662 100644
--- a/koji/__init__.py
+++ b/koji/__init__.py
@@ -1537,6 +1537,8 @@ name=build
     yc_parts.append("baseurl=%s\n" % urls[0])
     for url in urls[1:]:
         yc_parts.append("        %s\n" % url)
+    if opts.get('module_hotfixes'):
+        yc_parts.append("module_hotfixes=1\n")
     config_opts['yum.conf'] = ''.join(yc_parts)
 
     plugin_conf = {

